src/bundle.ts

```diff
@@ -0,0 +1,32 @@
+export default class Bundle {
+	public tarStream: NodeJS.ReadableStream
+
+	/**
+	 * deviceType: The slug of the device type that this bundle has been created
+	 * for
+	 */
+	public deviceType: string
+
+	/**
+	 * architecture: The architecture that this resin bundle is targeting
+	 */
+	public architecture: string
```

Comment:
Sounds like an interesting area to investigate, certainly. It'd be very interesting to generate type definitions from device data, and I think there's probably quite a few interesting applications once we have that.

Could do similar things from SBVR too, to generate a schema in TS that we can use to automatically keep ideas. Depends how much time we want to invest in this, and this all needs more time to work out exactly what value we're looking for, but there's a lot of very interesting scope here.